HW1/P6/P6.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import re
+
+import findspark
+findspark.init()
+from pyspark import SparkContext, SparkConf
+
+conf = SparkConf().setAppName('P4').setMaster('local')
+sc = SparkContext(conf=conf)
+
+# Implemnts a graph for a Markov chain based geenration of phrases based on word occurences
+
+def counter(l):
+	b = []
+	for i in range(len(l)):
+		a = (l[i], l.count(l[i]))
+		if a not in b:
+			b.append(a)
+	return b
+
+def getGraph(source):
+	lines = sc.textFile(source)
+	
+	words = lines.flatMap(lambda line: line.split())
+	words = words.filter(lambda word: word.isupper() == False)
+	words = words.filter(lambda word: word.isdigit() == False)
+	words = words.filter(lambda word: word[0:-1].isupper() == False or word[-1] == '.')
+
+	word_index = words.zipWithIndex()
+	index_word = word_index.map(lambda (word, index): (index, word))
+	index_plus_one_word = index_word.map(lambda (index, word): (index+1, word))
+	index_plus_two_word = index_word.map(lambda (index, word): (index+2, word))
+
+	words_three = index_plus_two_word.join(index_plus_one_word).join(index_word).sortByKey().persist()
+	words_three_no_index = words_three.map(lambda (a, ((b, c), d)): ((b, c), (d)))
+	words_two_list1 = words_three_no_index.groupByKey()
+	words_two_list2 = words_two_list1.map(lambda ((a, b,), c): ((a, b), list(c)))
+	words_two_list3 = words_two_list2.map(lambda ((a, b,), c): ((a, b), counter(c)))
+	return words_two_list3
+
+
+graph = getGraph('./shakespeare.txt')
+
+
+def highest_count(l):
```

Comment:
It seems that you don't choose a word according to its number of appearances (weighted distribution).